PublicGitArchive/pga-create/cmd/pga-create/select.go

```diff
@@ -1,200 +1,118 @@
-	StarsFile         string
-	LanguagesFile     string
-	ReposFile         string
-	FilteredLanguages []string
-	MinStars          int
-	TopN              int
-	URLTemplate       string
+	ReposFile   string
+	IndexList   string
+	MinStars    int
+	TopN        int
+	URLTemplate string
 }
 
-func selectRepos(parameters selectionParameters) {
-	spin := spinner.New(spinner.CharSets[11], 100*time.Millisecond)
-	spin.Writer = os.Stderr
-	spin.Start()
-
-	var selectedRepos map[int]bool
-	if len(parameters.FilteredLanguages) > 0 {
-		spin.Suffix = " reading " + parameters.LanguagesFile
-		selectedRepos = filterLanguages(parameters.LanguagesFile, parameters.FilteredLanguages)
-	}
-	spin.Suffix = " reading " + parameters.StarsFile
-	selectedRepos = filterStars(
-		parameters.StarsFile, parameters.MinStars, parameters.TopN, selectedRepos)
-	spin.Stop()
-	bar := pb.New(len(selectedRepos))
-	bar.Output = os.Stderr
-	bar.ShowFinalTime = true
-	bar.ShowPercent = false
-	bar.ShowSpeed = false
-	bar.SetMaxWidth(80)
-	bar.Start()
-	defer bar.Finish()
-	f, err := os.Open(parameters.ReposFile)
+func selectRepos(params selectionParameters) {
+	f, err := os.Open(params.ReposFile)
 	if err != nil {
-		fail("opening repositories file "+parameters.ReposFile, err)
+		fail("opening repositories file "+params.ReposFile, err)
 	}
 	defer f.Close()
+
 	gzf, err := gzip.NewReader(f)
 	if err != nil {
-		fail("decompressing repositories file "+parameters.ReposFile, err)
+		fail("decompressing repositories file "+params.ReposFile, err)
 	}
 	defer gzf.Close()
-	scanner := bufio.NewScanner(gzf)
-	var count int
-	for scanner.Scan() {
-		var repoID int
-		var repoName string
-		line := scanner.Text()
-		n, err := fmt.Sscan(line, &repoID, &repoName)
-		if err != nil || n != 2 {
-			if err == nil {
-				err = errors.New("failed to parse " + line)
-			}
-			fail("parsing repositories file "+parameters.ReposFile, err)
-		}
-		if selectedRepos[repoID] {
-			count++
-			bar.Increment()
-			fmt.Fprintf(os.Stdout, parameters.URLTemplate+"\n", repoName)
-		}
-		if count >= len(selectedRepos) {
-			break
+
+	var idxw *csv.Writer
+	if params.MinStars > 0 {
+		idxList, err := os.Create(params.IndexList)
+		if err != nil {
+			fail("creating file "+params.IndexList, err)
 		}
-	}
+		defer func() {
+			if err := idxList.Close(); err != nil {
+				fail("closing output file "+params.IndexList, err)
+			}
+		}()
 
-	if sErr := scanner.Err(); sErr != nil {
-		fail("scanning repositories file "+parameters.ReposFile, sErr)
-	}
-}
+		gzw := gzip.NewWriter(idxList)
+		defer func() {
+			if err := gzw.Close(); err != nil {
+				fail("closing output gz file "+params.IndexList, err)
+			}
+		}()
 
-func filterStars(path string, minStars int, topN int, selectedRepos map[int]bool) map[int]bool {
-	f, err := os.Open(path)
-	if err != nil {
-		fail("opening stars file "+path, err)
+		idxw = csv.NewWriter(gzw)
 	}
-	defer f.Close()
 
-	gzf, err := gzip.NewReader(f)
+	r := csv.NewReader(gzf)
+	// read headers
```

Comment:
This is maybe my taste but I would write the comment before the whole block and not in the middle of it:

```go
￼ 	// read headers
	r := csv.NewReader(gzf)
	_, err = r.Read()
	if err != nil {
		fail("parsing repositories file "+params.ReposFile, err)
	}
```

Reply:
Understood the comment incorrectly